src/main/java/seedu/address/logic/commands/EditTaskCommand.java

```diff
@@ -0,0 +1,183 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TASK_NOTE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TASK_TITLE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_TASKS;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.task.Note;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.Title;
+
+/**
+ * Edits the details of an existing task in the address book.
+ */
+public class EditTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "editTask";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the task identified "
+            + "by the index number used in the displayed task list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TASK_TITLE + "TITLE] "
+            + "[" + PREFIX_TASK_NOTE + "NOTE]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_TASK_TITLE + "Prepare Agenda "
+            + PREFIX_TASK_NOTE + "To book venue";
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the address book.";
+
+    private final Index index;
+    private final EditTaskDescriptor editTaskDescriptor;
+
+    /**
+     * @param index              of the task in the filtered task list to edit
```

Comment:
what happen to the javadoc here ? 
```suggestion
     * @param index of the task in the filtered task list to edit
```

Reply:
Formatter wants to align the description of the param descriptions